### PR TITLE
Checkout: Add basic wpcom-checkout package

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This adds a new npm package called `wpcom-checkout` which can be used as a toolbox for checkout in calypso (and possibly other places in the future). The advantages of moving code to this package is that such code will have strict type checking, as opposed to within calypso where types are allowed but not checked. It also provides a mechanism for that code to be shared with other WordPress.com checkout implementations outside of calypso (such as within wp-admin).

This PR moves only some line item translation functions to the package. These are functions which bridge the gap between the `shopping-cart` package and the `composite-checkout` package. Other code can be moved at a later time.

#### Testing instructions

This should not cause any visible changes to checkout, so mostly we just need to verify that nothing is broken. Some notable things to test which have been modified:

- Verify that translations for the words "Total", "Subtotal", and "Coupon" in the cart are still translated correctly to other locales.
- Verify that a purchase made through the PayPal Express payment method still works correctly.
- Verify that a purchase made through the existing card payment method still works correctly.
- Verify that when you have sufficient credits to cover a full transaction, you still only see the "Free" payment method.

### CODE REVIEW

File: client/my-sites/checkout/composite-checkout/test/translate-cart.js [hunk 1]
```diff
@@ -129,27 +121,6 @@ describe( 'translateResponseCartToWPCOMCart', function () {
 			} );
 		} );
 
-		describe( 'taxes', function () {
```
Comment (nbloomf) [inline client/my-sites/checkout/composite-checkout/test/translate-cart.js:132]: It looks like we're losing these tests, which makes me a little nervous. Is it possible/sensible to move them to the new package?

Comment (sirbrillig) [reply client/my-sites/checkout/composite-checkout/test/translate-cart.js:132]: That's a good point. They were testing the moved functions implicitly, which is why I didn't notice. I'll create some tests in the package.

Comment (sirbrillig) [reply client/my-sites/checkout/composite-checkout/test/translate-cart.js:132]: I did better and added tests to cover the specific cases of the functions.

**Comment by nbloomf** [comment]
Looks good! Translations work as expected:

image

I tested paypal, stored card, and full credits purchases with no issues.

**Review by nbloomf — APPROVED** [review]

**Pull request merged**